protocol/test/foundry/sun/Flood.t.sol

```diff
@@ -0,0 +1,576 @@
+    event SeasonOfPlentyField(uint256 toField);
+
+    function setUp() public {
+        initializeBeanstalkTestState(true, false);
+        address well = C.BEAN_ETH_WELL;
+        // init user.
+        farmers.push(users[1]);
+        vm.prank(farmers[0]);
+        C.bean().approve(BEANSTALK, type(uint256).max);
+
+        // Initialize well to balances. (1000 BEAN/ETH)
+        addLiquidityToWell(
+            well,
+            1000000e6, // 10,000 Beans
+            1000 ether // 10 ether.
+        );
+
+        addLiquidityToWell(
+            C.BEAN_WSTETH_WELL,
+            10000e6, // 10,000 Beans
+            10 ether // 10 ether.
+        );
+
+        season.siloSunrise(0);
+        season.siloSunrise(0);
+
+        // users 1 and 2 deposits 1000 beans into the silo.
+        address[] memory depositUsers = new address[](2);
+        depositUsers[0] = users[1];
+        depositUsers[1] = users[2];
+        depostBeansForUsers(depositUsers, 1_000e6, 10_000e6);
+        // depostBeansForUser(users[2], 1000e6);
+
+        // give user2 some eth
+        vm.deal(users[2], 10 ether);
+
+        // without this, 25 rainSunrises runs out of gas
+        vm.pauseGasMetering();
+    }
+
+    function testNotRaining() public {
+        Storage.Season memory s = seasonGetters.time();
+
+        assertFalse(s.raining);
+    }
+
+    function testRaining() public {
+        field.incrementTotalPodsE(1000e18);
+        season.rainSunrise();
+        bs.mow(users[1], C.BEAN);
+
+        Storage.Rain memory rain = seasonGetters.rain();
+        Storage.Season memory s = seasonGetters.time();
+
+        assertTrue(s.rainStart == s.current);
+        assertTrue(s.raining);
+        assertEq(rain.pods, bs.totalPods());
+        assertEq(rain.roots, 20008000000000000000000000);
+
+        SiloGettersFacet.AccountSeasonOfPlenty memory sop = siloGetters.balanceOfSop(users[1]);
+
+        assertTrue(sop.lastRain == s.rainStart);
+        assertTrue(sop.roots == 10004000000000000000000000);
+    }
+
+    function testStopsRaining() public {
+        field.incrementTotalPodsE(1000e18);
+        season.rainSunrise();
+        bs.mow(users[1], C.BEAN);
+
+        season.droughtSunrise();
+        bs.mow(users[1], C.BEAN);
+
+        Storage.Season memory s = seasonGetters.time();
+        assertTrue(s.rainStart == s.current - 1);
+
+        SiloGettersFacet.AccountSeasonOfPlenty memory sop = siloGetters.balanceOfSop(users[1]);
+        assertTrue(sop.lastRain == 0);
+    }
+
+    function testSopsWhenAtPeg() public {
+        season.rainSunrises(25);
+        Storage.Season memory s = seasonGetters.time();
+
+        assertTrue(s.lastSop == 0);
+        assertTrue(s.lastSopSeason == 0);
+    }
+
+    function testSopsBelowPeg() public {
+        setDeltaBforWell(-1000e6, C.BEAN_ETH_WELL, C.WETH);
+        // update pumps
+        updateMockPumpUsingWellReserves(C.BEAN_ETH_WELL);
```

Comment:
should be able to update the functions prior to not need these functions anymore